tests/CodeBehind/UnitTests/BuildTests.cs

```diff
@@ -100,14 +100,40 @@ IEnumerator IEnumerable.GetEnumerator()
 		}
 	}
 
+	sealed class TestProjectInfo
+	{
+		public string RootDirectory { get; }
+		public string OutputDirectory { get; }
+		public string ObjPath { get; }
+		public string BinPath { get; }
+		public string GeneratedPath { get; }
+		public string SlnPath { get; }
+		public string ProjectName { get; }
+		public string TestName { get; }
+
+		public TestProjectInfo (string projectName, string testName, string rootDirectory, string outputRootDir)
+		{
+			TestName = testName;
+			RootDirectory = rootDirectory;
+			ProjectName = projectName;
+
+			ObjPath = Path.Combine (rootDirectory, "obj");
+			GeneratedPath = Path.Combine (ObjPath, XABuildPaths.Configuration, "generated");
+			BinPath = Path.Combine (rootDirectory, "bin", XABuildPaths.Configuration);
```

Comment:
@grendello could you also put the test projects one folder deeper, such as `temp`? (Our MSBuild tests do this)

This would let us only have to archive the `temp` folder.

VSTS was having to do this and it took 16 minutes:
```
2018-09-04T19:14:36.4802948Z Uploading 4862 files
...
2018-09-04T19:31:32.9632583Z File upload succeed.
2018-09-04T19:31:32.9632946Z Upload 'E:\A\_work\14\a' to file container: '#/2397083/Artifacts'
2018-09-04T19:31:32.9633335Z Associated artifact 1741668 with build 2002748
```

Some of the files here we probably don't need archived:

![image](https://user-images.githubusercontent.com/840039/45054154-c04e5a00-b051-11e8-8656-1be79eb28e8f.png)

Reply:
@jonathanpeppers the test output and artifacts are all in `TestDebug/CodeBehind/` and its subdirs - can you use that?